### PR TITLE
fix: call Queue.isReady() before job promote is called

### PR DESCRIPTION
**Pull request by chocof**
The issue is described here: https://github.com/OptimalBits/bull/issues/1231

### CODE REVIEW

**Comment by gohiei** [comment]
I test it at my local machine, and it works well.

File: lib/job.js [hunk 1]
```diff
@@ -304,12 +304,13 @@ Job.prototype.moveToDelayed = function(timestamp, ignoreLock) {
...
   const queue = this.queue;
   const jobId = this.id;
-
-  return scripts.promote(queue, jobId).then(result => {
-    if (result === -1) {
-      throw new Error('Job ' + jobId + ' is not in a delayed state');
-    }
-  });
+  return queue.isReady().then(() =>
+    scripts.promote(queue, jobId).then(result => {
```
Comment (manast) [inline lib/job.js]: a return is needed here so that the promise returned by promote is propagated correctly.

Comment (chocof) [reply lib/job.js]: Hello @manast ,  do you mean the following part of the code
`... .then(() => scripts.promote() ...)`
if yes then in es6 the syntax `() => value` is identical to `() => { return value }` so it is practically like writing 
`return queue.isReady().then(() => { return scripts.promote(queue, jobId) }).then() ... `
If this is not what you meant please correct me

Comment (chocof) [reply lib/job.js]: however if you would like to use the previous syntax `() => {return ...}` (which as i can see is used in this file) then i can edit my pr

Comment (manast) [reply lib/job.js]: no sorry, the code is correct, I did not notice the arrow function.

**Pull request merged**